### ISSUE TITLE
Another M1 build error - "_EVP_MD_get_size" "Undefined symbols for architecture arm64" Build error on openssl library

### ISSUE THREAD

**Issue by darrenfu** [facebookincubator/velox#3431]
### Bug description

Recently I resync-ed the main branch, and I got the following build error, when I run either the main function in `velox/exec/tests/VeloxIn10MinDemo.cpp` or run any unit test class, e.g. `velox/functions/prestosql/tests/ArrayHasDuplicatesTest.cpp`. 

This issue has blocked my further development. I used to have a dirty workaround to force CMake to load the arm64 openssl library (`/opt/homebrew/Cellar/openssl@1.1/1.1.1s`) instead of the x86-64 one (`/opt/homebrew/Cellar/openssl@1.1/1.1.1d`) when I ran into other openssl arm64 library errors, it doesn't work for this sepcific error.  
Need expert help!
```bash
sudo rm -f  /opt/homebrew/opt/openssl@1.1 && ln -s /opt/homebrew/Cellar/openssl@1.1/1.1.1s /opt/homebrew/opt/openssl@1.1
```

### System information

Velox System Info v0.0.2
Commit: 7ed6f5f113b612353fc6b52236f57fef6a1f47cc
CMake Version: 3.25.0
System: Darwin-21.6.0
Arch: arm64
C++ Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
C++ Compiler Version: 14.0.0.14000029
C Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
C Compiler Version: 14.0.0.14000029
CMake Prefix Path: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk/usr;/opt/homebrew;/usr/local;/usr;/;/opt/homebrew/Cellar/cmake/3.25.0;/usr/local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local
  

### Relevant logs

```bash
https://gist.github.com/darrenfu/5c4b224a6306bdb431aeab81e52da2af
```

**Comment by darrenfu** [comment]
See the solution in https://github.com/facebookincubator/velox/issues/3436#issuecomment-1344871649

**Comment by majetideepak** [comment]
I hit this too and my workaround is 
```
brew unlink openssl;
brew link openssl@1.1
```
We should fix the build to pick up openssl@1.1 without these additional steps.
